### PR TITLE
add unit test for jobset webhook updates

### PR DESCRIPTION
**Pull request by kannon92**
Add unit test for updates.

### CODE REVIEW

File: api/jobset/v1alpha2/jobset_webhook_test.go [hunk 1]
```diff
@@ -747,3 +747,132 @@ func TestValidateCreate(t *testing.T) {
...
+			oldJs: &JobSet{
+				ObjectMeta: validObjectMeta,
+				Spec: JobSetSpec{
+					Suspend:        ptr.To(true),
+					ReplicatedJobs: validReplicatedJobs,
+				},
+			},
+		},
+		{
+			name: "forbid update on manged-by",
```
Comment (danielvegamyhre) [inline api/jobset/v1alpha2/jobset_webhook_test.go:814]: typo in "managed-by"

Comment (kannon92) [reply api/jobset/v1alpha2/jobset_webhook_test.go]: done.

**Comment by danielvegamyhre** [comment]
Thanks for improving the test coverage here. Let's merge this after the complex validation PR is checked in, I think merge conflicts will be easier to resolve that way than the reverse

**Comment by kannon92** [comment]
> Thanks for improving the test coverage here. Let's merge this after the complex validation PR is checked in, I think merge conflicts will be easier to resolve that way than the reverse

Sounds good.

I was thinking about https://github.com/kubernetes-sigs/jobset/issues/463 and I was going to add unit tests for that code. Decided to move it to a separate PR so we can at least document better that elastic jobs are not supported yet.

**Comment by kannon92** [comment]
/hold

Waiting on kubectl-validate code.

**Comment by kannon92** [comment]
> Thanks for improving the test coverage here. Let's merge this after the complex validation PR is checked in, I think merge conflicts will be easier to resolve that way than the reverse

I think we can merge this now.

**Comment by kannon92** [comment]
@danielvegamyhre I think we can get this in now. PTAL

**Pull request merged**